### PR TITLE
Deprecate empty offsets in get_path_collection_extents

### LINKED ISSUE #1735: `_path.get_path_collection_extents` potentially wrong return value
`_path.get_path_collection_extents`, when applied to a collections with `offsets=[]` treats the offsets as if they were [0, 0](this behavior is used in path.get_paths_extents).

This interpretation of an empty collection is error-prone, and caused #1490. Hence it would be desirable to decide what should `path.get_path_collection_extents` return when given an empty collection, and have `path.get_path_extents` not rely on current behavior.

This issue is created to not lose track of the bug, while closing #1490 and merging #1497.

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Closes #1735

Based on the issue, the outcome of an empty offsets is not well defined. This is now deprecated.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by timhoffm** [comment]
> Running the tests locally does not seem to break anything, but there may be image comparisons tests that fails because of this 

Codecov says your addition is never exercised in the tests; so it cannot break the tests :smile: .

**Review by timhoffm — DISMISSED** [review]
Anyway, I wouldn't know what to test. Passing empty offsets is trivial and almost not worth a test. But maybe we should do that to pin this as intended behavior.

**Comment by oscargus** [comment]
What I meant was that we do not have any tests that relies on being able to pass an empty `offsets` (nor an empty `paths`). And, as it turned out, no in the additional ones on the CI (as I may not have all optional dependencies installed).

The problem I think is to know if it ever is useful to pass an empty `offsets` or if some valid input data would pass an empty `offsets`. Testing the function directly doesn't really make sense just to exercise the code path, but one would probably like to do it from an actual user-level API (even though this technically is one).

**Comment by timhoffm** [comment]
If we are not sure that somebody is relying on the current but ambiguous behavior, we should deprecate it instead of immediately erroring out. That gives people time to object. The issue has been standing since 2013, so there is no rush in turning it into an error.

**Comment by anntzer** [comment]
*Perhaps* what this should really return is a null bbox (see discussion at https://github.com/matplotlib/matplotlib/pull/17115#issuecomment-1153001448)?  Although we can always change that later if we manage to resolve #17115 at some point.

**Comment by oscargus** [comment]
Not sure why I put this as draft, but rebased and updated deprecation version.

**Review by timhoffm — APPROVED** [review]

File: doc/api/next_api_changes/deprecations/23200-OG.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Calling ``paths.get_path_collection_extents`` with empty ``offsets``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Calling  `~.get_path_collection_extents` with an empty ``offsets`` parameter
```
Comment (QuLogic) [inline doc/api/next_api_changes/deprecations/23200-OG.rst:4]:
```suggestion
Calling ``paths.get_path_collection_extents`` with empty *offsets*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Calling  `~.get_path_collection_extents` with an empty *offsets* parameter
```

**Review by ksunden — APPROVED** [review]

**Pull request merged**